### PR TITLE
Fix lazy loading

### PR DESCRIPTION
**Pull request by rjs-picturepark**

### CODE REVIEW

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.html [hunk 1]
```diff
@@ -1,11 +1,24 @@
-<div class="image-container" [class.content-item-thumbnail--list-view]="isListView" (lazyload)="markAsVisible()"
+<!-- <div class="image-container" [class.content-item-thumbnail--list-view]="isListView" (lazyload)="markAsVisible()"
```
Comment (neotek) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.html:1]: Remove comments.

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.ts [hunk 2]
```diff
@@ -2,10 +2,11 @@ import { Component, OnChanges, SimpleChanges, SecurityContext, OnInit, Input, In
...
 import { SafeUrl, SafeHtml, DomSanitizer } from '@angular/platform-browser';
 import { NON_VIRTUAL_CONTENT_SCHEMAS_IDS, BROKEN_IMAGE_URL } from '../../../utilities/constants';
-import { switchMap } from 'rxjs/operators';
+import { switchMap, map, tap } from 'rxjs/operators';
 import { BaseBrowserItemComponent } from '../browser-item-base/browser-item-base.component';
 import { ThumbnailSize, Content, ShareDetail, ShareContentDetail } from '@picturepark/sdk-v1-angular';
 import { ContentService, fetchContentById } from '@picturepark/sdk-v1-angular';
+import { Observable } from 'rxjs/internal/Observable';
 
 @Component({
```
Comment (neotek) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.ts:11]: Use OnPush change detection

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.ts [hunk 3]
```diff
@@ -49,19 +50,21 @@ export class ContentItemThumbnailComponent extends BaseBrowserItemComponent<Cont
...
         const output = content.outputs.find(i => i.outputFormatId === 'Thumbnail' + this.thumbnailSize);
-        if (output) {
-          this.thumbnailUrl = this.sanitizer.bypassSecurityTrustResourceUrl(output.viewUrl!);
-        } else {
-          this.thumbnailUrl = this.sanitizer.bypassSecurityTrustResourceUrl(content.iconUrl!);
+        this.isLoading = true;
+        if ( (output && output.viewUrl) || content.iconUrl) {
+          const thumbnailSubscription = this.loadItem.subscribe(() => {
+            this.isLoading = false;
+            this.thumbnailUrl = this.trust( output ? output.viewUrl : content.iconUrl);
```
Comment (neotek) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.ts:62]: Output is always defined.

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.html [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
+  <img *ngIf="!isLoading" class="content-item-thumbnail__image" [ngClass]="{'shadow': shadow}" [src]="thumbnailUrl"
     [alt]="'ContentBrowser.NoImageForItem' | pptranslate" (error)="updateUrl($event)" />
 </div>
 <ng-template #virtualContent>
-  <div [ngClass]="{'shadow': shadow}" class="virtual-item-container">
-    <div class="virtual-item" [innerHTML]="virtualItemHtml"></div>
+  <div [ngClass]="{'shadow': shadow}" class="content-item-thumbnail__image-virtual-item-container">
+    <div class="content-item-thumbnail__image-virtual-item-display" [innerHTML]="virtualItemHtml"></div>
+  </div>
+</ng-template> -->
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.html:11]: This code block is still commented. Should it be removed?

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.scss [hunk 5]
```diff
@@ -1,38 +1,40 @@
...
-    justify-content: center;
-    overflow: hidden;
-
-    position: absolute;
-    top: 0;
-    right: 0;
-    left: 0;
-    bottom: 0;
+    &__image-virtual-item-container {
+        // height: 100%;
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.scss:22]: Do we need these commented properties?

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.ts [hunk 6]
```diff
@@ -2,10 +2,11 @@ import { Component, OnChanges, SimpleChanges, SecurityContext, OnInit, Input, In
 
 import { SafeUrl, SafeHtml, DomSanitizer } from '@angular/platform-browser';
 import { NON_VIRTUAL_CONTENT_SCHEMAS_IDS, BROKEN_IMAGE_URL } from '../../../utilities/constants';
-import { switchMap } from 'rxjs/operators';
+import { switchMap, map, tap } from 'rxjs/operators';
```
Comment (tc4rdpp) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.ts:5]: Are map and tap being used?

File: src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.ts [hunk 7]
```diff
@@ -43,23 +44,25 @@ export class ContentItemThumbnailComponent extends BaseBrowserItemComponent<Cont
...
     super(injector);
   }
 
   async ngOnInit() {
-
     if (this.shareItem) {
       const content = this.shareItem.contentSelections.find(i => i.id === this.item.id);
 
       if (content) {
         const output = content.outputs.find(i => i.outputFormatId === 'Thumbnail' + this.thumbnailSize);
```
Comment (rjs-picturepark) [inline src/picturepark-sdk-v1-angular/projects/picturepark-sdk-v1-angular-ui/src/lib/shared-module/components/content-item-thumbnail/content-item-thumbnail.component.ts:58]: Check with Urs that the viewUrl sometimes doesn't come from the ouputs

**Pull request merged**